lib/rubygems/core_ext/kernel_require.rb

```diff
@@ -43,18 +43,18 @@ def require(path)
     # https://github.com/rubygems/rubygems/pull/1868
     resolved_path = begin
       rp = nil
-      $LOAD_PATH[0...Gem.load_path_insert_index || -1].each do |lp|
-        safe_lp = lp.dup.tap(&Gem::UNTAINT)
-        begin
-          if File.symlink? safe_lp # for backward compatibility
-            next
+      Gem.suffixes.each do |s|
```

Comment:
Note: I confirmed `Gem.suffixes` returned `["", ".rb", ".bundle"]` on macOS.

Reply:
Thanks! I had two doubts here:

* Is the "empty extension" actually needed?
* Should I explicitly loop first through ".rb", and then through the rest of the extensions, so that we don't depend on the order of `Gem.suffixes`?

Reply:
Both of the issues are actually present in the current code, so I'll move this forward for now.